src/cargo/util/toml/mod.rs

```diff
@@ -1068,7 +1107,11 @@ pub struct TomlProject {
     #[serde(rename = "forced-target")]
     forced_target: Option<String>,
     links: Option<String>,
+    #[serde(default)]
+    #[serde(deserialize_with = "workspace_vec_string")]
     exclude: Option<MaybeWorkspace<Vec<String>>>,
+    #[serde(default)]
+    #[serde(deserialize_with = "workspace_vec_string")]
     include: Option<MaybeWorkspace<Vec<String>>>,
     publish: Option<MaybeWorkspace<VecStringOrBool>>,
```

Comment:
This field is also affected by this `MaybeWorkspace` untagged enum. It will never get the inner error.

Reply:
Hi rustin, which field is mentioned here? Do you mean the `publish` field?

Reply:
`publish` should not use `#[serde(deserialize_with = "maybe_workspace_vec_string")]` as it would need to output a different error message. 

This brings into question all error messages for `MaybeWorkspace`. It looks like the error message should be customized for each place that uses `MaybeWorkspace`. That seems outside of the scope of this PR and something that should be addressed later. @epage (or one of the other maintainers) might have a different idea so I would wait for them to comment.

Reply:
If we can't generalize `MaybeWorkspace`'s deserializer, should we consider removing it to force people to customize it like this?

Reply:
I think it can be made generic but it might take a little bit of work to do so. I was mostly trying to say that we should limit the scope of this PR to only fields that are `MaybeWorkspace<Vec<String>>`.

Reply:
Well, the PR is ready except for this and some improvement is better than no improvement, so I'm going to go ahead and merge and create a broader issue.